### PR TITLE
Dev/rackscale allocator

### PR DESCRIPTION
**Pull request by gz**

### CODE REVIEW

**Review by gz — COMMENTED** [review]
@hunhoffe maybe check if my additions are good, then we can merge it?

File: lib/rpc/src/rpc.rs [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
+use core2::io::Result as IOResult;
+use core2::io::Write;
 
 /// Node ID for servers/clients
-pub type NodeId = u64;
+pub type ClientId = u64;
+
+#[derive(Debug)]
+pub(crate) struct ClientIdRes {
+    // abomonate doesn't work well with type aliases so use
```
Comment (hunhoffe) [inline lib/rpc/src/rpc.rs:14]: TODO(erika): fix comment

**Review by hunhoffe — COMMENTED** [review]
LGTM barring a few small changes I'll try to push later today!

File: kernel/src/arch/x86_64/rackscale/processops/release_physical.rs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+    let mut shmem_managers = SHMEM_MANAGERS.lock();
+    // TODO: here and in alloc should percolate error to client
+    let manager = shmem_managers[node_id as usize]
+        .as_mut()
+        .expect("Error - no shmem manager found for client");
+
+    // TODO: we don't have real frame information, so skip actual calls
+    let ret = if frame_size <= BASE_PAGE_SIZE as u64 {
+        //manager.release_base_page(frame)
```
[the remark this answers is not in the document]
Comment (gz) [reply kernel/src/arch/x86_64/rackscale/processops/release_physical.rs:123]: Ah yes, we should get proper info here I think I did adjust the client side but not the controller side

File: lib/rpc/src/rpc.rs (hunk 1 above)
Comment (hunhoffe) [reply lib/rpc/src/rpc.rs:14]: Done

File: kernel/src/arch/x86_64/rackscale/processops/allocate_physical.rs [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+            let frame = Frame::new(PAddr::from(frame_base), size as usize, affinity as usize);
+            let fid = NrProcess::<Ring3Process>::allocate_frame_to_process(pid, frame)?;
+
+            // Add frame mapping to local map
+            {
+                let mut frame_map = FRAME_MAP.write();
+                frame_map
+                    .try_reserve(1)
+                    .map_err(|_e| RPCError::InternalError)?;
+                info!(
```
[the remark this answers is not in the document]
Comment (hunhoffe) [reply kernel/src/arch/x86_64/rackscale/processops/allocate_physical.rs:83]: Done

File: kernel/src/arch/x86_64/rackscale/processops/release_physical.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (hunhoffe) [reply kernel/src/arch/x86_64/rackscale/processops/release_physical.rs]: fixed!

**Comment by hunhoffe** [comment]
@gz should be ready for merge now! I wanted to give you chance to review since I made changes

**Pull request merged**